crates/node/src/networking/p2p/pea2pea.rs

```diff
@@ -165,34 +168,55 @@ impl P2P {
             peer_list.clone()
         };
 
-        protocol::Handshake::V1(protocol::HandshakeV1 {
+        protocol::internal::Handshake {
             my_p2p_listen_addr,
             peers,
             http_port: self.http_port,
-        })
+        }
     }
 
     async fn process_diagnostics_request(
         &self,
         source: SocketAddr,
-        req: protocol::DiagnosticsRequestKind,
+        req: protocol::internal::DiagnosticsRequestKind,
     ) -> io::Result<()> {
-        let resp = protocol::Message::V0(protocol::MessageV0::DiagnosticsResponse(
-            self.public_node_key,
-            protocol::DiagnosticsResponseV0::Version {
-                major: env!("CARGO_PKG_VERSION_MAJOR").parse::<u16>().unwrap(),
-                minor: env!("CARGO_PKG_VERSION_MINOR").parse::<u16>().unwrap(),
-                patch: env!("CARGO_PKG_VERSION_PATCH").parse::<u16>().unwrap(),
-                build: format!(
-                    "{}: {}",
-                    env!("VERGEN_BUILD_TIMESTAMP"),
-                    env!("VERGEN_GIT_DESCRIBE")
-                ),
-            },
-        ));
+        let resp = match req {
+            protocol::internal::DiagnosticsRequestKind::Version => {
+                protocol::internal::Message::DiagnosticsResponse(
+                    self.public_node_key,
+                    protocol::internal::DiagnosticsResponse::Version {
+                        major: env!("CARGO_PKG_VERSION_MAJOR").parse::<u16>().unwrap(),
+                        minor: env!("CARGO_PKG_VERSION_MINOR").parse::<u16>().unwrap(),
+                        patch: env!("CARGO_PKG_VERSION_PATCH").parse::<u16>().unwrap(),
+                        build: format!(
+                            "{}: {}",
+                            env!("VERGEN_BUILD_TIMESTAMP"),
+                            env!("VERGEN_GIT_DESCRIBE")
+                        ),
+                    },
+                )
+            }
+            protocol::internal::DiagnosticsRequestKind::Resources => {
+                let (cpus, mem, gpus) = self.node_resources;
+                protocol::internal::Message::DiagnosticsResponse(
+                    self.public_node_key,
+                    protocol::internal::DiagnosticsResponse::Resources { cpus, mem, gpus },
+                )
+            }
 
-        let bs =
-            Bytes::from(bincode::serialize(&resp).expect("diagnostics response serialization"));
+            protocol::internal::DiagnosticsRequestKind::Metrics => {
+                let data = metrics::export_metrics();
+                protocol::internal::Message::DiagnosticsResponse(
+                    self.public_node_key,
+                    protocol::internal::DiagnosticsResponse::Metrics(data),
+                )
+            }
+        };
+
+        let bs = Bytes::from(
+            protocol::new_serialize_msg(self.protocol_version, resp)
+                .expect("diagnostics response serialization"),
+        );
```

Comment:
Nodes never communicate between each other with diagnostics messages. Hence we don't need to support the original protocol here.